### DISCUSSION TITLE
What does `closest .python-version` mean with global python enabled?

### DISCUSSION THREAD

**Discussion by ifsheldon** [astral-sh/rye#1214] [Q&A]
In the documentation, there's

> Afterwards if you run python outside of a Rye managed project it will spawn a Python interpreter that is shipped with Rye. It will honor the closest .python-version file for you

but what does it mean to be `closest`? Is it the most recently used one? How can I configure it to behave like a conda base env? i.e., inside a rye project, use the project's python and anywhere else use the global python by default.

**Comment by zanieb** [comment]
Usually "closest" in this context means we'll look in parent directories (of the working directory) until we find a `.python-version` file.

**Reply by ifsheldon** [reply]
Awesome! Then, in theory, I can init my home directory as a rye project, then I can have a default python managed by rye for my daily one-off code, right? Or, can it be simpler?
